src/java.desktop/windows/native/libjsound/PLATFORM_API_WinOS_DirectSound.cpp

```diff
@@ -258,6 +258,8 @@ INT32 DAUDIO_GetDirectAudioDeviceDescription(INT32 mixerIndex, DirectAudioDevice
         DS_unlockCache();
         return FALSE;
     }
+    if(FAILED(::CoInitialize(NULL)))
+        return false;
```

Comment:
DS_unlockCache if the CoInitialize fails?

Reply:
Thank you for pointing this, I have clubbed CoInitialize check along with other condition. In case of failure it will return there itself.

Reply:
This code has the same issue: "if (!DS_lockCache() || FAILED(::CoInitialize(NULL)))"